src/network.rs

```diff
@@ -25,6 +25,7 @@ const MAX_RETRIES: u8 = 8;
 const RETRY_CODES: [u16; 4] = [429, 500, 502, 504];
 
 /// This struct is cast to `c_void`, then to `PROJ_NETWORK_HANDLE` so it can be passed around
+#[repr(C)]
```

Comment:
I'm curious about the motivation for making this `repr(C)`.

I *think* this struct is only used within rust and it's only interacted with externally as an opaque pointer. So I thought it was unnecessary.

Looking at [the docs](https://doc.rust-lang.org/nomicon/other-reprs.html#reprc):
>  Any type you expect to pass through an FFI boundary should have repr(C)

Which... I'm not sure what to make of. If we're only passing a *pointer* to the struct, maybe it's fine to omit?

In any case, I don't expect this change to break anything, but curious to learn more.

Reply:
I am quite clueless about the nuance here. Probably best to ask someone…

Reply:
If I'm reading [this](https://stackoverflow.com/a/30742180/416626) correctly, we don't need the `repr`, because – as you correctly point out – we're just passing around the pointer, _not_ modifying its fields in `libproj`.